rotkehlchen/chain/substrate/manager.py

```diff
@@ -0,0 +1,555 @@
+            )
+
+        manager = args[0]
+        if len(manager.available_node_attributes_map) == 0:
+            raise RemoteError(f'{manager.chain} has no nodes available')
+
+        kwargs_ = kwargs.copy()
+        requested_nodes = []
+        for node, node_attributes in manager._get_nodes_call_order():
+            kwargs_.update({'node_interface': node_attributes.node_interface})
+            try:
+                result = func(manager, *args[1:], **kwargs_)
+            except (RemoteError, SubstrateRequestException, ValueError):
+                requested_nodes.append(str(node))
+                log.warning(
+                    f'{manager.chain} {func.__name__!r} failed to request via '
+                    f'{node} node at endpoint {node.endpoint()}',
+                )
+                continue
+
+            return result
+
+        raise RemoteError(
+            f'{manager.chain} failed to request after trying the following nodes: '
+            f'{", ".join(requested_nodes)}',
+        )
+    return wrapper
+
+
+class SubstrateManager():
+    def __init__(
+            self,
+            chain: Chain,
+            greenlet_manager: GreenletManager,
+            msg_aggregator: MessagesAggregator,
+    ) -> None:
+        """An interface to any Substrate chain supported by Rotki.
+
+        It uses Polkascan py-substrate-interface for interacting with the
+        substrate blockchains and the Subscan API as a chain explorer.
+
+        Official substrate chains documentation:
+        https://substrate.dev/rustdocs/v2.0.0/sc_service/index.html
+        https://guide.kusama.network/docs/en/kusama-index
+        https://wiki.polkadot.network/en/
+
+        Polkascan py-scale-codec:
+        https://github.com/polkascan/py-scale-codec/tree/master
+
+        Polkascan py-substrate-interface:
+        https://github.com/polkascan/py-substrate-interface
+        https://polkascan.github.io/py-substrate-interface/base.html
+
+        Postman Substrate Interface API:
+        https://documenter.getpostman.com/view/9969999/SWT5iLXH?version=latest
+
+        Subscan API documentation:
+        https://docs.api.subscan.io
+        """
+        if chain not in Chain:
+            raise AttributeError(f'Unexpected SubstrateManager chain: {chain}')
+
+        log.debug(f'Initializing substrate {chain} manager')
+        self.chain = chain
+        self.greenlet_manager = greenlet_manager
+        self.msg_aggregator = msg_aggregator
+        self.available_node_attributes_map: DictNodeNameNodeAttributes = {}
+        self.chain_properties: ChainProperties
+        self._connect()
+
+    def _check_chain_id(self, node_interface: NodeInterface) -> None:
+        """Validate a node connects to the expected chain.
+        """
+        # Check connection and chain ID
+        try:
+            chain_id = self._get_chain_id(node_interface=node_interface)
+        except RemoteError as e:
+            log.error(str(e))
```

Comment:
Why do this? If you want to log the error, especially without any additional comment just do it at the lowest point where it is raised. And here do nothing and let it bubble up.